conn.go

```diff
@@ -656,18 +670,34 @@ func (w *writeCoalescer) Write(p []byte) (int, error) {
 }
 
 func (w *writeCoalescer) writeFlusher(interval time.Duration, quit chan struct{}) {
-	ticker := time.NewTicker(interval)
-	defer ticker.Stop()
+	timer := time.NewTimer(interval)
+	defer timer.Stop()
 	defer w.flush()
 
+	if !timer.Stop() {
+		<-timer.C
+	}
+
 	for {
+		// wait for a write to start the flush loop
+		w.fcond.L.Lock()
+		for !w.running {
+			w.fcond.Wait()
```

Comment:
if the connection has been closed without any `Write` this goroutine will be waiting here forever

Reply:
Good spot, do you want to submit a PR? Think it should be sufficient to broadcast if not running?

Reply:
Sure, i can do it. 

I also experienced an issue while i was running this version of gocql, with a deadlock always waiting for this line and the `Write` calls stuck in L660. I haven't been able to reproduce it in a test in a consistent way.

Reply:
and there's now #1200 for the same issue, so clearly reproduced.